### PR TITLE
LB check with failover and session expiration test

### PR DESCRIPTION
**Pull request by martin-kanis**

### CODE REVIEW

File: provision/rosa-cross-dc/keycloak-benchmark-crossdc-tests/src/test/java/org/keycloak/benchmark/crossdc/AbstractCrossDCTest.java [hunk 1]
```diff
@@ -113,5 +120,27 @@ public void tearDownTestEnvironment() {
                 });
 
         MOCK_COOKIE_MANAGER.getCookieStore().removeAll();
+
+        DC_1.kc().markLBCheckUp();
+        DC_2.kc().markLBCheckUp();
+        waitUntilPrimaryIsUp(LOAD_BALANCER_KEYCLOAK.getKeycloakServerUrl(), DC_1.getKeycloakServerURL(), DC_2.getKeycloakServerURL());
+    }
+
+    public void waitForFailover(String clientUrl, String primaryUrl, String backupUrl) throws IOException, InterruptedException {
```
Comment (kami619) [inline provision/rosa-cross-dc/keycloak-benchmark-crossdc-tests/src/test/java/org/keycloak/benchmark/crossdc/AbstractCrossDCTest.java:129]: How would this end if the failover doesnt happen for some reason ?

Comment (martin-kanis) [reply provision/rosa-cross-dc/keycloak-benchmark-crossdc-tests/src/test/java/org/keycloak/benchmark/crossdc/AbstractCrossDCTest.java:129]: Failover should finish when AWS health check detects that a site is down (refreshed every 30s) + when DNS records are updated (depends on DNS caching setup). It usually takes around 1.5 - 2 minutes.  
When failover fails for whatever reason the test will probably hang there. The easiest thing we can do is to add `@Timeout(value = 10, unit = TimeUnit.MINUTES)`  annotation on the test.

Comment (kami619) [reply provision/rosa-cross-dc/keycloak-benchmark-crossdc-tests/src/test/java/org/keycloak/benchmark/crossdc/AbstractCrossDCTest.java]: I think thats a good idea either way, lets do that to add a timeout at the testcase level @martin-kanis

File: provision/rosa-cross-dc/keycloak-benchmark-crossdc-tests/src/test/java/org/keycloak/benchmark/crossdc/client/KeycloakClient.java [hunk 2]
```diff
@@ -218,6 +224,28 @@ public void markLBCheckUp() throws URISyntaxException, IOException, InterruptedE
...
+    public void waitToBeActive(KeycloakClient loadBalancer) throws UnknownHostException, InterruptedException {
+        int startTime = Time.currentTime();
+        int timeLimit = startTime + 600; // 10 minutes
+        LOG.infof("Waiting for Keycloak %d to be active.", keycloakServerUrl);
+        while (!isActive(loadBalancer) && Time.currentTime() <= timeLimit) {
+            Thread.sleep(1000);
+        }
+
+        if (Time.currentTime() > timeLimit) {
+            throw new RuntimeException("Keycloak " + keycloakServerUrl + " did not become active in 10 minutes.");
```
Comment (kami619) [inline provision/rosa-cross-dc/keycloak-benchmark-crossdc-tests/src/test/java/org/keycloak/benchmark/crossdc/client/KeycloakClient.java:245]: thanks for this change @mhajas

**Review by kami619 — APPROVED** [review]
LGTM

**Pull request merged**